### DISCUSSION TITLE
Presets for Date Range Picker

### DISCUSSION THREAD

**Discussion by davidchalifoux** [mantinedev/mantine#5428] [Feature requests]
I would LOVE if there was a built-in way to provide presets for the date range picker. I'm thinking something similar to Antd: https://ant.design/components/date-picker#components-date-picker-demo-preset-ranges

![Screenshot 2023-12-15 at 1 37 21 PM](https://github.com/mantinedev/mantine/assets/3289533/dde86934-4fcb-4f49-ad47-ca340c7ea78a)

**Comment by davidchalifoux** [comment]
Here's a quick implementation of this idea: https://github.com/mantinedev/mantine/compare/master...davidchalifoux:mantine:master
Let me know your thoughts @rtivital

**Reply by catnipan** [reply]
I suggest allowing the Calendar to support `leftSection?: React.ReactNode`, similar to how the Button does. usage would look like `leftSection={<Presets .../>}`, this providing greater flexibility and composability.

**Comment by rtivital** [comment]
This feature is planned to be added, but not in nearest releases

**Comment by bastiengrignon** [comment]
@rtivital Is this feature is planned to be implemented soon ? 
If not a quick way to allow people to do that would be to allow access to the children inside `popoverProps` to be able to override it. https://github.com/mantinedev/mantine/blob/c087030bbc67a895cb39475b0494ed5a6dc5e7ab/packages/%40mantine/dates/src/components/PickerInputBase/PickerInputBase.tsx#L38

Do you think that removing the **Omit** could be possible of it's not a good thing to do ?
